File: benchmarks/fair/intloop.c
#include <stdio.h>
#include <stdlib.h>
int main(void){
    const char*e=getenv("BENCH_N"); long long n=e?atoll(e):0; if(n<=0)n=50000000LL;
    long long t=1;
    for(long long i=0;i<n;i++) t=(t*31+i)%1000000007LL;
    printf("%lld\n",t); return 0;
}

File: benchmarks/fair/intloop.go
package main
import ("fmt";"os";"strconv")
func main(){
    n:=int64(50000000)
    if v,err:=strconv.ParseInt(os.Getenv("BENCH_N"),10,64); err==nil&&v>0 { n=v }
    var t int64=1
    for i:=int64(0);i<n;i++ { t=(t*31+i)%1000000007 }
    fmt.Println(t)
}
